ui/contexts/currencyRate.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { getSelectedNetworkClientId } from '../selectors';
+import useCurrencyRatePolling from '../hooks/useCurrencyRatePolling';
+
+export const CurrencyRateProvider = ({ children }) => {
+  const selectedNetworkClientId = useSelector(getSelectedNetworkClientId);
```

Comment:
[nit] why is this state retrieved here and passed in as an arg while `useCurrencyRateCheck` and `completedOnboarding` are retrieved directly in the `useCurrencyRatePolling` hook?

Reply:
Since this PR is meant to be a stepping stone towards fully UI driven currency rate polling, I figure that when global polling via the `CurrencyRateProvider` gets replaced in favor of more granular polling in the actual child components, we will want the networkClientId to be a configurable param of `useCurrencyRatePolling`.

Are you suggesting that `selectedNetworkClientId` is used as the default value in `useCurrencyRatePolling` if no networkClientId arg is provided?

Reply:
yeah that makes sense. Yeah perhaps we can make it a default instead. Not a big deal either way. But this is fine as is as well.

Reply:
addressed here https://github.com/MetaMask/metamask-extension/pull/23476/commits/b038d113828de7b8efd91f4efebc59c837eacb9f